moderation_test.go

```diff
@@ -92,3 +93,187 @@ func TestGetBannedUsers(t *testing.T) {
+			&BanUserParams{BroadcasterID: "1234", ModeratorId: "5678", Body: BanUserRequestBody{
+				UserId:   "9876",
+				Duration: 300,
+				Reason:   "no reason",
+			}},
+			`{"error":"Bad Request","status": 400,"message":"user is already banned"}`,
+			"user is already banned",
+		},
+		{
+			http.StatusOK,
+			&Options{ClientID: "my-client-id", UserAccessToken: "moderator-access-token"},
+			&BanUserParams{BroadcasterID: "1234", ModeratorId: "5678", Body: BanUserRequestBody{
+				UserId:   "9876",
+				Duration: 300,
+				Reason:   "no reason",
+			}},
+			`{"data": [{"broadcaster_id": "1234","moderator_id": "5678","user_id": "9876","created_at": "2021-09-28T19:22:31Z","end_time": "2021-09-28T19:27:31Z"}]}`,
+			"",
+		},
+		{
+			http.StatusOK,
+			&Options{ClientID: "my-client-id", UserAccessToken: "moderator-access-token"},
+			&BanUserParams{BroadcasterID: "1234", ModeratorId: "5678", Body: BanUserRequestBody{
+				UserId: "9876",
+				Reason: "no reason",
+			}},
+			`{"data": [{"broadcaster_id": "1234","moderator_id": "5678","user_id": "9876","created_at": "2021-09-28T18:22:31Z","end_time": null}]}`,
+			"",
+		},
+	}
+
+	for _, testCase := range testCases {
+		c := newMockClient(testCase.options, newMockHandler(testCase.statusCode, testCase.respBody, nil))
+
+		resp, err := c.BanUser(testCase.params)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if resp.StatusCode != testCase.statusCode {
+			t.Errorf("expected status code to be %d, got %d", testCase.statusCode, resp.StatusCode)
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			if resp.ErrorStatus != testCase.statusCode {
+				t.Errorf("expected error status to be \"%d\", got \"%d\"", testCase.statusCode, resp.ErrorStatus)
+			}
+
+			if resp.ErrorMessage != testCase.expectedErrMsg {
+				t.Errorf("expected error message to be \"%d\", got \"%d\"", testCase.statusCode, resp.ErrorStatus)
+			}
+		}
+
+		if len(resp.Data.Bans) != 0 {
+			if resp.Data.Bans[0].BoardcasterId != testCase.params.BroadcasterID {
+				t.Errorf("expected broadcaster id to be \"%s\", got \"%s\"", testCase.params.BroadcasterID, resp.Data.Bans[0].BoardcasterId)
+			}
+
+			if resp.Data.Bans[0].ModeratorId != testCase.params.ModeratorId {
+				t.Errorf("expected moderator id to be \"%s\", got \"%s\"", testCase.params.ModeratorId, resp.Data.Bans[0].ModeratorId)
+			}
+
+			if resp.Data.Bans[0].UserId != testCase.params.Body.UserId {
+				t.Errorf("expected user id to be \"%s\", got \"%s\"", testCase.params.Body.UserId, resp.Data.Bans[0].UserId)
+			}
+
+			if resp.Data.Bans[0].EndTime != "" {
+				layout := "2006-01-02T15:04:05Z"
+				createdTime, _ := time.Parse(layout, resp.Data.Bans[0].CreatedAt)
+				endTime, _ := time.Parse(layout, resp.Data.Bans[0].EndTime)
+
+				expireTime := createdTime.Add(time.Duration(testCase.params.Body.Duration * int(time.Second)))
+
+				if !expireTime.Equal(endTime) {
+					t.Errorf("expected endtime to be \"%s\", got \"%s\"", expireTime, resp.Data.Bans[0].EndTime)
+				}
```

Comment:
These lines can then be written like so:

```suggestion
				expireTime := resp.Data.Bans[0].CreatedAt.Add(time.Duration(testCase.params.Body.Duration * int(time.Second)))

				if !expireTime.Equal(resp.Data.Bans[0].EndTime) {
					t.Errorf("expected endtime to be \"%s\", got \"%s\"", expireTime, resp.Data.Bans[0].EndTime)
				}
```